onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc

```diff
@@ -120,10 +120,14 @@ Status ResizeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   std::vector<float> scales_hw;
   std::vector<int32_t> sizes_hw;
   std::vector<int32_t> axes;
-  const bool isNhwc = model_builder.GetPreferredLayout() == DataLayout::NHWC;
+  const bool is_nhwc = model_builder.GetPreferredLayout() == DataLayout::NHWC;
```

Comment:
I would expect you still need to check the domain here (or infer the layout from the sizes/scales values) as the preferred layout would be constant but the node's layout and domain will change.  

On the first call the node is in ONNX domain and NCHW. After the EP says it wants the node, if the EP's preferred layout is NHWC the layout transformation occurs, and the node moves to the internal NHWC domain.

Reply:
But this `AddToModelBuilderImpl` will only be called once after the layout transformation, during graph compilation.